cartoframes/context.py

```diff
@@ -663,6 +776,114 @@ def sync(self, dataframe, table_name):
         """
         pass
 
+    # TODO: replace query with this method
+    def querycopy(self, query, table_name=None, decode_geom=False):
+        """Pull the result from an arbitrary SQL query from a CARTO account
+        into a pandas DataFrame. Can also be used to perform database
+        operations (creating/dropping tables, adding columns, updates, etc.).
```

Comment:
Technically this allows for `CREATE TABLE ...`, `ALTER TABLE ...`, `DROP TABLE ..`, `UPDATE ...` since it's just a query against the database. To me it seems counterproductive to prevent users from using it for this purpose. However, in the docs, I could point out it's better to use `CartoContext.sql_client` (which is just an instance of the carto python sdk's SQL Client class), which is json-returning instead of `pandas.DataFrame`-returning

Reply:
https://www.postgresql.org/docs/current/static/sql-copy.html

> query
> A SELECT, VALUES, INSERT, UPDATE or DELETE command whose results are to be copied. Note that parentheses are required around the query.
>
> For INSERT, UPDATE and DELETE queries a RETURNING clause must be provided, and the target relation must not have a conditional rule, nor an ALSO rule, nor an INSTEAD rule that expands to multiple statements

Reply:
👍 Great, I'll add that information to the documentation and refer people to the the sql client instead.

Reply:
I was still in the old SQL mindset and didn't think through if these would actually work, so I'm really glad you shared the postgres docs